src/esp/gfx/PbrShader.h

```diff
@@ -185,28 +190,91 @@ class PbrShader : public Magnum::GL::AbstractShaderProgram {
      * will use the node's semantic ID
      */
     InstancedObjectId = (1 << 12) | ObjectId,
+
+    /**
+     * Has ClearCoat layer.
+     */
+    ClearCoatLayer = 1 << 13,
+    // TODO : possibly break out each layer's flags to a separate flags state
+    // machine?
+    /**
+     * Has ClearCoat Texture in ClearCoat layer
+     */
+    CCLayer_CCTexture = (1 << 14) | ClearCoatLayer,
+    /**
+     * Has Roughness Texture in ClearCoat layer
+     */
+    CCLayer_RoughnessTexture = (1 << 15) | ClearCoatLayer,
```

Comment:
These could be merged into one, `ClearCoatFactorRoughnessTexture`, as per my suggestion with the swizzles above.

Similarly for the code that was already there, it doesn't make sense to have separate `MetallicTexture` and `RoughnessTexture` as there's already `NoneRoughnessMetallicTexture`. I doubt these variants were ever sufficiently tested given there's basically no assets that wouldn't have a combined R/M texture, so just drop them.